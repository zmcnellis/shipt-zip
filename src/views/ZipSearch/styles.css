/* Grid styles ---------------------------------------------------------------*/
.cards {
  display: grid;
  grid-template-columns: repeat(1, minmax(250px, 1fr));
  grid-gap: 30px;
}

/* Buttons -------------------------------------------------------------------*/
.button {
  display: inline-block;
  height: 48px;
  font-size: 11px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-decoration: none;
  width: 120px;
  line-height: 48px;
  font-weight: 700;
  text-align: center;
  box-shadow: 0 5px 25px rgba(0, 0, 0, 0.05);
  border-radius: 2px;
}

.button.button-primary {
  background-color: #71b045;
  color: #fff;
}

.button.button-lg {
  width: 320px;
}

.button.button-primary:hover {
  background-color: #669933;
  transition: all 0.3s ease;
  box-shadow: none;
}

/* Cards ---------------------------------------------------------------------*/
.cards {
  max-width: 960px;
  margin: 0 auto 30px;
}

.card-header {
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  max-width: 550px;
}

/* Search results ------------------------------------------------------------*/
.results-container {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 0 auto;
  max-width: 1080px;
  justify-content: space-around;
}

.results-header {
  text-align: center;
  text-transform: uppercase;
  font-size: 11px;
}

.button-area {
  text-align: center;
  padding: 20px 0;
}

/* Media queries -------------------------------------------------------------*/
@media (min-width: 700px) {
  /* search results */
  .results-container {
    flex-direction: row;
  }
}

@media (min-width: 1000px) {
  /* cards */
  .cards {
    grid-template-columns: repeat(3, minmax(250px, 1fr));
  }
}
