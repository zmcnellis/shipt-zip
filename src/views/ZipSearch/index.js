import React, { Component } from 'react';
import './styles.css';
import Announcement from '../../components/Announcement';
import Button from '../../components/Button';
import Card from '../../components/Card';
import Footer from '../../components/Footer';
import Header from '../../components/Header';
import SearchResult from '../../components/SearchResult';
import Hero from '../../components/Hero';
import * as api from '../../services/ZipApi'

// main view for the search page
class ZipSearch extends Component {
  // store api results in state
  constructor(props) {
    super(props);
    this.state = {
      stores: [],
      hasError: false
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.setSearchReference = this.setSearchReference.bind(this);
    this.getStartedClick = this.getStartedClick.bind(this);
  }

  // set the error state to true for validation errors
  displayError() {
    this.setState({
      hasError: true
    });
  }

  // call zip code api when search button is clicked
  handleSubmit(zipCode) {
    // check if the zip code is in a valid format (5 digits)
    if (api.isValidZip(zipCode) === false) {
      this.displayError();
    }
    else {
      api.getNearestStores(zipCode)
        .then(result => {
          this.setState({
            stores: result,
            hasError: false
          });
        })
        .catch(error => {
          this.displayError();
        });
    }
  }

  // keep track of input box reference in order to set focus on button clicks
  setSearchReference(ref) {
    this.searchBox = ref;
  }

  // set the focus of the input box when a "Get Started" button is clicked
  getStartedClick() {
    this.searchBox.focus();
  }

  // render search results after search if no errors
  renderSearchResults() {
    let hasSearchResults = this.state.stores.length > 0;
    let searchResults = !hasSearchResults ? "" : this.state.stores.sort(api.comparator).map(store => <SearchResult id={store.id} name={store.name} date={store.launch_date}/>);

    if (hasSearchResults) {
      return (
        <section id="searchResults">
          <h4 className="results-header">Delivering From:</h4>
          <div className="results-container">{searchResults}</div>
          <div className="button-area">
            <Button text="Get Started" size="large" onClick={this.getStartedClick}/>
          </div>
        </section>
      );
    }
  }

  // render components
  render() {
    return (
      <div>
        <Announcement/>
        <Header/>

        <main>
          <Hero handleSubmit={this.handleSubmit} hasError={this.state.hasError} setSearchReference={this.setSearchReference}/>
          {this.renderSearchResults()}

          <h1 className="card-header">Everything you love about Target at your fingertips.</h1>

          <section className="cards">
            <Card title="Scroll through the aisles." description="Create an order from your local store in our app or on our website." step="1"/>
            <Card title="Our shoppers work their magic." description="Connect with your shopper and get live updates from the aisles." step="2"/>
            <Card title="We deliver your groceries." description="Schedule a delivery, and your order will arrive in as soon as 1 hour." step="3"/>
          </section>

          <div className="button-area">
            <Button text="Get Started" size="large" onClick={this.getStartedClick}/>
          </div>
        </main>

        <Footer/>
      </div>
    );
  }
}

export default ZipSearch;
