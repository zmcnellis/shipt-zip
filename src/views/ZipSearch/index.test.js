import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ZipSearch from '.';

describe('ZipSearch', () => {
  it('should match snapshot', () => {
    const output = shallow(
      <ZipSearch/>
    );
    expect(output).toMatchSnapshot();
  });
});
