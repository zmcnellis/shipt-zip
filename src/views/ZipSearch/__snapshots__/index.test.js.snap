// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipSearch should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ZipSearch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Announcement />,
        <Header />,
        <main>
          <Hero
                    handleSubmit={[Function]}
                    hasError={false}
                    setSearchReference={[Function]}
          />
          <h1
                    className="card-header"
          >
                    Everything you love about Target at your fingertips.
          </h1>
          <section
                    className="cards"
          >
                    <Card
                              description="Create an order from your local store in our app or on our website."
                              step="1"
                              title="Scroll through the aisles."
                    />
                    <Card
                              description="Connect with your shopper and get live updates from the aisles."
                              step="2"
                              title="Our shoppers work their magic."
                    />
                    <Card
                              description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                              step="3"
                              title="We deliver your groceries."
                    />
          </section>
          <div
                    className="button-area"
          >
                    <Button
                              onClick={[Function]}
                              size="large"
                              text="Get Started"
                    />
          </div>
</main>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Hero
              handleSubmit={[Function]}
              hasError={false}
              setSearchReference={[Function]}
/>,
            undefined,
            <h1
              className="card-header"
>
              Everything you love about Target at your fingertips.
</h1>,
            <section
              className="cards"
>
              <Card
                            description="Create an order from your local store in our app or on our website."
                            step="1"
                            title="Scroll through the aisles."
              />
              <Card
                            description="Connect with your shopper and get live updates from the aisles."
                            step="2"
                            title="Our shoppers work their magic."
              />
              <Card
                            description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                            step="3"
                            title="We deliver your groceries."
              />
</section>,
            <div
              className="button-area"
>
              <Button
                            onClick={[Function]}
                            size="large"
                            text="Get Started"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleSubmit": [Function],
              "hasError": false,
              "setSearchReference": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Everything you love about Target at your fingertips.",
              "className": "card-header",
            },
            "ref": null,
            "rendered": "Everything you love about Target at your fingertips.",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Card
                  description="Create an order from your local store in our app or on our website."
                  step="1"
                  title="Scroll through the aisles."
/>,
                <Card
                  description="Connect with your shopper and get live updates from the aisles."
                  step="2"
                  title="Our shoppers work their magic."
/>,
                <Card
                  description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                  step="3"
                  title="We deliver your groceries."
/>,
              ],
              "className": "cards",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "description": "Create an order from your local store in our app or on our website.",
                  "step": "1",
                  "title": "Scroll through the aisles.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "description": "Connect with your shopper and get live updates from the aisles.",
                  "step": "2",
                  "title": "Our shoppers work their magic.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "description": "Schedule a delivery, and your order will arrive in as soon as 1 hour.",
                  "step": "3",
                  "title": "We deliver your groceries.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                onClick={[Function]}
                size="large"
                text="Get Started"
/>,
              "className": "button-area",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "size": "large",
                "text": "Get Started",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Announcement />,
          <Header />,
          <main>
            <Hero
                        handleSubmit={[Function]}
                        hasError={false}
                        setSearchReference={[Function]}
            />
            <h1
                        className="card-header"
            >
                        Everything you love about Target at your fingertips.
            </h1>
            <section
                        className="cards"
            >
                        <Card
                                    description="Create an order from your local store in our app or on our website."
                                    step="1"
                                    title="Scroll through the aisles."
                        />
                        <Card
                                    description="Connect with your shopper and get live updates from the aisles."
                                    step="2"
                                    title="Our shoppers work their magic."
                        />
                        <Card
                                    description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                                    step="3"
                                    title="We deliver your groceries."
                        />
            </section>
            <div
                        className="button-area"
            >
                        <Button
                                    onClick={[Function]}
                                    size="large"
                                    text="Get Started"
                        />
            </div>
</main>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Hero
                handleSubmit={[Function]}
                hasError={false}
                setSearchReference={[Function]}
/>,
              undefined,
              <h1
                className="card-header"
>
                Everything you love about Target at your fingertips.
</h1>,
              <section
                className="cards"
>
                <Card
                                description="Create an order from your local store in our app or on our website."
                                step="1"
                                title="Scroll through the aisles."
                />
                <Card
                                description="Connect with your shopper and get live updates from the aisles."
                                step="2"
                                title="Our shoppers work their magic."
                />
                <Card
                                description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                                step="3"
                                title="We deliver your groceries."
                />
</section>,
              <div
                className="button-area"
>
                <Button
                                onClick={[Function]}
                                size="large"
                                text="Get Started"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleSubmit": [Function],
                "hasError": false,
                "setSearchReference": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Everything you love about Target at your fingertips.",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Everything you love about Target at your fingertips.",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Card
                    description="Create an order from your local store in our app or on our website."
                    step="1"
                    title="Scroll through the aisles."
/>,
                  <Card
                    description="Connect with your shopper and get live updates from the aisles."
                    step="2"
                    title="Our shoppers work their magic."
/>,
                  <Card
                    description="Schedule a delivery, and your order will arrive in as soon as 1 hour."
                    step="3"
                    title="We deliver your groceries."
/>,
                ],
                "className": "cards",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "description": "Create an order from your local store in our app or on our website.",
                    "step": "1",
                    "title": "Scroll through the aisles.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "description": "Connect with your shopper and get live updates from the aisles.",
                    "step": "2",
                    "title": "Our shoppers work their magic.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "description": "Schedule a delivery, and your order will arrive in as soon as 1 hour.",
                    "step": "3",
                    "title": "We deliver your groceries.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  onClick={[Function]}
                  size="large"
                  text="Get Started"
/>,
                "className": "button-area",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClick": [Function],
                  "size": "large",
                  "text": "Get Started",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
